### PR TITLE
Single GPU `DADCSampler`

### PR DESCRIPTION
**Pull request by ordabayevy**

### CODE REVIEW

**Review by ordabayevy — COMMENTED** [review]
@mbabadi can review this PR for a single GPU sampler?

File: scvid/data/sampler.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+from typing import Iterator, Sequence
+
+import torch
+from torch.utils.data.sampler import Sampler
+
+
+class DADCSampler(Sampler):
```
Comment (mbabadi) [inline scvid/data/sampler.py:7]: When you implement multi GPU, do you plan to keep this class as well? If so, perhaps call it `DistributedAnnDataCollectionSingleConsumerSampler`? :D

File: test/test_sampler.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        collate_fn=collate_fn,
+    )
+
+    # iterate through dataloader
+    list(data_loader)
+
+    # assert one epoch
+    assert sampler.epoch == 1
+
+    if num_workers > 0:
```
Comment (mbabadi) [inline test/test_sampler.py:83]: nice!

**Review by mbabadi — CHANGES_REQUESTED** [review]
Nice PR! I put in a few small comments.

**Review by ordabayevy — COMMENTED** [review]
@mbabadi can you have another look?

**Review by mbabadi — APPROVED** [review]

**Comment by ordabayevy** [comment]
I forgot to commit renaming changes in the `test` folder. Hopefully it should pass now.

**Pull request merged**